hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestMORDataSource.scala

```diff
@@ -614,4 +614,66 @@ class TestMORDataSource extends HoodieClientTestBase {
       .load(basePath)
     assertEquals(N + 1, hoodieIncViewDF1.count())
   }
+
+  @ParameterizedTest
+  @ValueSource(booleans = Array(true, false))
+  def testMORPartitionPrune(partitionEncode: Boolean): Unit = {
+    val partitions = Array("2021/03/01", "2021/03/02", "2021/03/03", "2021/03/04", "2021/03/05")
```

Comment:
does the hive style partitioning `year=2021/xxx` work as well?

Reply:
Yes, the hive style partition is supported.

Reply:
should we add a test for hive style as well?

Reply:
Nice suggestion! done!